public/app/features/alerting/unified/hooks/useProduceNewRuleGroup.tsx

```diff
@@ -133,16 +99,183 @@ export function usePauseRuleInGroup() {
+ */
+export function useMoveRuleGroup() {
+  const [produceNewRuleGroup] = useProduceNewRuleGroup();
+  const [fetchRuleGroup] = alertRuleApi.endpoints.getRuleGroupForNamespace.useLazyQuery();
+  const [upsertRuleGroup] = alertRuleApi.endpoints.upsertRuleGroupForNamespace.useMutation();
+  const [deleteRuleGroup] = alertRuleApi.endpoints.deleteRuleGroupFromNamespace.useMutation();
+
+  // @TODO maybe add where we moved it from and to for additional peace of mind
+  const successMessage = t('alerting.rule-groups.move.success', 'Successfully moved rule group');
+
+  return useAsync(
+    async (ruleGroup: RuleGroupIdentifier, namespaceName: string, groupName?: string, interval?: string) => {
+      // we could technically support moving rule groups to another folder, though we don't have a "move" wizard yet.
+      if (isGrafanaRulesSource(ruleGroup.dataSourceName)) {
+        throw new Error('Moving a Grafana-managed rule group to another folder is currently not supported.');
+      }
+
+      const action = moveRuleGroupAction({ namespaceName, groupName, interval });
+      const { newRuleGroupDefinition, rulerConfig } = await produceNewRuleGroup(ruleGroup, action);
+
+      const oldNamespace = ruleGroup.namespaceName;
+      const targetNamespace = action.payload.namespaceName;
+
+      const oldGroupName = ruleGroup.groupName;
+      const targetGroupName = action.payload.groupName;
+
+      const isGroupRenamed = Boolean(targetGroupName) && oldGroupName !== targetGroupName;
+
+      // if we're also renaming the group, check if the target does not already exist
+      if (targetGroupName && isGroupRenamed) {
+        const targetGroup = await fetchRuleGroup({
+          rulerConfig,
+          namespace: targetNamespace,
+          group: targetGroupName,
+          // since this could throw 404
+          requestOptions: { showErrorAlert: false },
+        })
+          .unwrap()
+          .catch(notFoundToNullOrThrow);
+
+        if (targetGroup?.rules?.length) {
+          throw new Error('Target group already has rules, merging rule groups is currently not supported.');
+        }
+      }
+
+      // create the new group in the target namespace
+      // ⚠️ it's important to do this before we remove the old group – better to have two groups than none if one of these requests fails
+      await upsertRuleGroup({
+        rulerConfig,
+        namespace: targetNamespace,
+        payload: newRuleGroupDefinition,
+        requestOptions: { successMessage },
+      }).unwrap();
+
+      // now remove the old one
+      const result = await deleteRuleGroup({
+        rulerConfig,
+        namespace: oldNamespace,
+        group: oldGroupName,
+        requestOptions: { showSuccessAlert: false },
+      }).unwrap();
+
+      return result;
+    }
   );
+}
+
+/**
+ * Rename a rule group but keep it within the same namespace, throws if the action targets an existing rule group
+ */
+export function useRenameRuleGroup() {
```

Comment:
I would find another name for this hook, as we are updating the interval. And the description could also mention this. Same for the `renameRuleGroupAction`

Reply:
I've struggled with this naming a bit, every action on a rule group allows for updating the interval so I thought of not including that in the name of the hook.

I think I'll keep it like this for now until we run into a scenario where some actions on rule groups do not allow for the interval to be updated.

I'll add it to the description though because it can't be inferred without reading the source 👌